orchagent/fdborch.cpp

```diff
@@ -418,108 +671,359 @@ void FdbOrch::doTask(NotificationConsumer& consumer)
+        {
+            /* Duplicate Mac */
+            SWSS_LOG_NOTICE("FdbOrch: mac=%s %s port=%s type=%s origin=%d is duplicate", entry.mac.to_string().c_str(), vlan.m_alias.c_str(), port_name.c_str(), type.c_str(), origin);
+            return true;
+        }
+        else if(origin != oldOrigin)
+        {
+            /* Mac origin has changed */
+            if((oldType == "static") && (oldOrigin == FDB_ORIGIN_PROVISIONED))
+            {
+                /* old mac was static and provisioned, it can not be changed by Remote Mac */
+                SWSS_LOG_NOTICE("Already existing static MAC:%s in Vlan:%d. "
+                        "Received same MAC from peer:%s; "
+                        "Peer mac ignored",
+                        entry.mac.to_string().c_str(), vlan.m_vlan_info.vlan_id,
+                        remote_ip.c_str());
+
+                return true;
+            }
+            else if((oldType == "static") && (oldOrigin == FDB_ORIGIN_VXLAN_ADVERTIZED) && (type == "dynamic"))
+            {
+                /* old mac was static and received from remote, it can not be changed by dynamic locally provisioned Mac */
+                SWSS_LOG_NOTICE("Already existing static MAC:%s in Vlan:%d "
+                        "from Peer:%s. Now same is provisioned as dynamic; "
+                        "Provisioned dynamic mac is ignored",
+                        entry.mac.to_string().c_str(), vlan.m_vlan_info.vlan_id,
+                        it->second.remote_ip.c_str());
+                return true;
+            }
+            else if(oldOrigin == FDB_ORIGIN_VXLAN_ADVERTIZED)
+            {
+                if((oldType == "static") && (type == "static"))
+                {
+                    SWSS_LOG_WARN("You have just overwritten existing static MAC:%s "
+                            "in Vlan:%d from Peer:%s, "
+                            "If it is a mistake, it will result in inconsistent Traffic Forwarding",
+                            entry.mac.to_string().c_str(),
+                            vlan.m_vlan_info.vlan_id,
+                            it->second.remote_ip.c_str());
+                }
+            }
+        }
+        else /* (origin == oldOrigin) */
+        {
+            /* Mac origin is same, all changes are allowed */
+            /* Allowed
+             * Bridge-port is changed or/and
+             * Sticky bit from remote is modified or
+             * provisioned mac is converted from static<-->dynamic
+             */
+        }
+
+        macUpdate = true;
+    }
+
     sai_attribute_t attr;
     vector<sai_attribute_t> attrs;
 
     attr.id = SAI_FDB_ENTRY_ATTR_TYPE;
-    attr.value.s32 = (type == "dynamic") ? SAI_FDB_ENTRY_TYPE_DYNAMIC : SAI_FDB_ENTRY_TYPE_STATIC;
+    if (origin == FDB_ORIGIN_VXLAN_ADVERTIZED)
+    {
+        attr.value.s32 = (type == "dynamic") ? SAI_FDB_ENTRY_TYPE_STATIC_MACMOVE : SAI_FDB_ENTRY_TYPE_STATIC;
```

Comment:
where is the define `SAI_FDB_ENTRY_TYPE_STATIC_MACMOVE`. The SAI [PR1024](https://github.com/opencomputeproject/SAI/pull/1024) doesn't include this constant

Reply:
This is how it was proposed initially;
later as part of PR filing it was changed; Once the PR1024 gets merged; I will update the code;

Reply:
Code is updated as per SAI [PR1024 ](https://github.com/opencomputeproject/SAI/pull/1024)